Android/G3MAndroidSDK/src/org/glob3/mobile/specific/SQLiteStorage_Android.java

```diff
@@ -135,14 +145,74 @@ public void run(final G3MContext context) {
       }
    }
 
+    private boolean writeFileToCacheFolder(final String fileName, final byte[] content) {
+        final File f2 = new File(new File(_documentsDirectory),  fileName);
+        FileOutputStream fout = null;
+        try {
+            fout = new FileOutputStream(f2);
+            fout.write(content);
+            fout.flush();
+            return true;
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+            return false;
+        } catch (IOException e) {
+            e.printStackTrace();
+            return false;
+        } finally {
+            if (fout != null) {
+                try {
+                    fout.close();
+                } catch (IOException e1) {
+                    e1.printStackTrace();
+                }
+            }
+        }
+    }
+
+    private byte[] readFileFromCacheFolder(final String fileName) {
+        final File f2 = new File(new File(_documentsDirectory), fileName);
+        FileInputStream fin = null;
+        try {
+            fin = new FileInputStream(f2);
+            final int avail = fin.available();
+            if (avail == 0) {
```

Comment:
Thank you very much @DiegoGomezDeck! Looks very much like you are right. I will consult with @DrakkLord and one of us will fix this.